src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -96,7 +96,7 @@ public async Task<AuthenticationResult> RunAsync(CancellationToken cancellationT
                     LogReturnedToken(authenticationResult);
                     UpdateTelemetry(sw, apiEvent, authenticationResult);
                     LogMetricsFromAuthResult(authenticationResult, AuthenticationRequestParameters.RequestContext.Logger);
-                    LogSuccessfulTelemetryToClient(authenticationResult, telemetryEventDetails, telemetryClients);
+                    LogSuccessfulTelemetryToClient(authenticationResult, AuthenticationRequestParameters.RequestContext.TelemetryDatapoints, telemetryEventDetails, telemetryClients);
```

Comment:
Can you consolidate all these telemetry objects? I don't think we want to have both datapoints and eventdetails.

Reply:
The data points is what is exposed to the cache for IDWeb/other SDKs. I figured it would be better to keep the object in case we wanted to expand the data we request from the cache.

Reply:
I suggest consolidating them. This detail should not bleed throughout all our codebase.